linchpin/cli/__init__.py

```diff
@@ -258,28 +259,39 @@ def _write_distilled_context(self, run_data):
             except Exception:
                 pass
 
+        self.distilled_data = dist_data
+
         with open(context_file, 'w+') as f:
             f.write(json.dumps(dist_data))
 
 
-    def lp_down(self, pinfile, targets=(), run_id=None):
+    def distill_data(self, return_code, run_data):
         """
-        This function takes a list of targets, and performs a shutdown on
-        nodes in the target's topology. Only providers which support shutdown
-        from their API (Ansible) will support this option.
+        Distill data to the file <workspace>/resources/linchpin.distilled
+        from RunDB using the get_run_data method
 
-        CURRENTLY UNIMPLEMENTED
+        :param return_code:
+            Return code from the transaction
 
-        .. seealso:: lp_destroy
+        :param run_data:
+            return data, in a compressed format, which is used to
+            obtain all of the run_data on a per target basis
 
-        :param pinfile:
-            Provided PinFile, with available targets,
 
-        :param targets:
-            A tuple of targets to provision.
         """
 
-        pass
+        # Export distilled data in useful ways
+        # # Write out run_data to a file for now
+        if return_code:
+            distill_on_error = self.get_cfg('lp',
+                                            'distill_on_error',
+                                            default='False')
+            if ast.literal_eval(distill_on_error.title()):
+                self._write_distilled_context(run_data)
+        else:
+                self._write_distilled_context(run_data)
+
+        self.distilled_data = True
```

Comment:
I'm really confused.  line 262 above has this set to dist_data.

Reply:
This is remnants of some other code. I'll clean it up shortly.